### PR TITLE
feat(appBar): adds AppBar component :sparkles:

### PR DESCRIPTION
**Pull request by PlabanJr**

### CODE REVIEW

File: src/components/composites/Button/index.tsx [hunk 1]
```diff
@@ -40,19 +39,19 @@ type IconProps = {
...
-{
-  label?: string;
-  block?: boolean;
-  outline?: boolean;
-  transaprent?: boolean;
-  variant?: "critical" | "caution" | "positive" | "neutral" | "info" | "promote";
-  labelStyle?: TextStyle;
 };
 
+type ButtonProps = IconProps &
```
Comment (sankhadeeproy007) [inline src/components/composites/Button/index.tsx:44]: Are these just formatting changes?

File: src/components/composites/Button/index.tsx [hunk 2]
```diff
@@ -129,16 +128,16 @@ const Button = ({
     ...(outline || transaprent ? { color: variant ?? "indigo.6" } : {}),
   };
 
+  const { icon } = props;
```
Comment (sankhadeeproy007) [inline src/components/composites/Button/index.tsx:131]: Destructure icon in the component definition please.

File: src/components/composites/Button/index.tsx (hunk 1 above)
Comment (PlabanJr) [reply src/components/composites/Button/index.tsx:44]: Yes

File: src/components/composites/Button/index.tsx (hunk 2 above)
Comment (PlabanJr) [reply src/components/composites/Button/index.tsx]: Done

**Pull request merged**